resilience4j-hedge/src/main/java/io/github/resilience4j/hedge/HedgeConfig.java

```diff
@@ -0,0 +1,197 @@
+/*
+ *
+ *  Copyright 2021: Matthew Sandoz
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+package io.github.resilience4j.hedge;
+
+
+import io.github.resilience4j.hedge.internal.NamingThreadFactory;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.util.concurrent.*;
+
+/**
+ * HedgeConfig manages the configuration of Hedges
+ */
+public class HedgeConfig implements Serializable {
+
+    private static final long serialVersionUID = 2203981592465761602L;
+
+    private static final String HEDGE_DURATION_MUST_NOT_BE_NULL = "HedgeDuration must not be null";
+    private final MetricsType metricsType;
+    private final boolean shouldUseFactorAsPercentage;
+    private final int hedgeTimeFactor;
+    private final boolean shouldMeasureErrors;
+    private final int windowSize;
+    private final Duration cutoff;
+    private final ScheduledExecutorService hedgeExecutor;
+
+    private HedgeConfig(MetricsType metricsType, boolean shouldUseFactorAsPercentage, int hedgeTimeFactor, boolean shouldMeasureErrors, int windowSize, Duration cutoff, ScheduledExecutorService executor) {
+        this.metricsType = metricsType;
+        this.shouldUseFactorAsPercentage = shouldUseFactorAsPercentage;
+        this.hedgeTimeFactor = hedgeTimeFactor;
+        this.shouldMeasureErrors = shouldMeasureErrors;
+        this.windowSize = windowSize;
+        this.cutoff = cutoff;
+        this.hedgeExecutor = executor;
+    }
+
+    /**
+     * Returns a builder to create a custom HedgeConfig.
+     *
+     * @return a {@link HedgeConfig.Builder}
+     */
+    public static Builder custom() {
+        return new Builder();
+    }
+
+    public static Builder from(HedgeConfig baseConfig) {
+        return new Builder(baseConfig);
+    }
+
+    /**
+     * Creates a default Hedge configuration.
+     *
+     * @return a default Hedge configuration.
+     */
+    public static HedgeConfig ofDefaults() {
+        return new Builder().build();
+    }
+
+    public HedgeMetrics newMetrics() {
```

Comment:
seems like I need to move HedgeMetrics out - maybe the pattern has changed since I started? I can look into the pattern of keeping the metrics names in the hedge and pushing the rest out to the -metrics module. given that the bulk of the metrics is in a separate module with a lot of dependencies, I agree we should not depend on it from hedge if I move the metrics out. that would be a cycle.

Reply:
The pattern is `CircuitBreaker.Metrics`, `Retry.Metrics`, ...
We have a Metrics interface is in the Interface of the decorator.

Reply:
The CircuitBreakerConfig or RetryConfig doesn't know anything about metrics.

Reply:
these should all be separated now. I've also proposed some specific metrics